sql_generators/firefox_desktop_use_counters/templates/query.sql

```diff
@@ -0,0 +1,130 @@
+    {% for use_counter in firefox_desktop_use_counters %}
+      SUM(metrics.counter.{{use_counter.name}}) AS {{use_counter.name}},
+    {% endfor %}
+  FROM
+    `moz-fx-data-shared-prod.firefox_desktop.use_counters`
+  WHERE
+    DATE(submission_timestamp) = @submission_date
+  GROUP BY
+    1,
+    2,
+    3,
+    4
+  HAVING
+    COUNT(DISTINCT(client_info.client_id)) >= 5000
+),
+firefox_desktop_pivoted_raw AS (
+  SELECT
+    *
+  FROM
+    firefox_desktop_use_counts_by_day_version_and_country_stg a UNPIVOT(
+      cnt FOR metric IN (
+        {% for use_counter in firefox_desktop_use_counters %}
+          {{use_counter.name}}
+
+          {% if not loop.last %},
+          {% endif %}
+        {% endfor %}
+      )
+    )
+),
+firefox_desktop_staging AS (
+  SELECT
+    submission_date,
+    version_major,
+    country,
+    platform,
+    use_counter_content_documents_destroyed,
+    use_counter_top_level_content_documents_destroyed,
+    use_counter_service_workers_destroyed,
+    use_counter_shared_workers_destroyed,
+    use_counter_dedicated_workers_destroyed,
+    metric,
+    cnt,
+    CASE
+      WHEN metric LIKE 'use_counter_css_doc_%'
+        THEN SAFE_DIVIDE(cnt, use_counter_content_documents_destroyed)
+      ELSE NULL
+    END AS doc_rate,
+    CASE
+      WHEN metric LIKE 'use_counter_css_page_%'
+        THEN SAFE_DIVIDE(cnt, use_counter_top_level_content_documents_destroyed)
+      ELSE NULL
+    END AS page_rate,
+    CASE
+      WHEN metric LIKE 'use_counter_worker_service_%'
+        THEN SAFE_DIVIDE(cnt, use_counter_service_workers_destroyed)
+      ELSE NULL
+    END AS service_rate,
+    CASE
+      WHEN metric LIKE 'use_counter_worker_shared_%'
+        THEN SAFE_DIVIDE(cnt, use_counter_shared_workers_destroyed)
+      ELSE NULL
+    END AS shared_rate,
+    CASE
+      WHEN metric LIKE 'use_counter_worker_dedicated_%'
+        THEN SAFE_DIVIDE(cnt, use_counter_dedicated_workers_destroyed)
+      ELSE NULL
+    END AS dedicated_rate
+  FROM
+    firefox_desktop_pivoted_raw
+)
+SELECT
+  submission_date,
+  SAFE_CAST(version_major AS INT64) AS version_major,
+  country,
+  platform,
+  use_counter_content_documents_destroyed,
+  use_counter_top_level_content_documents_destroyed,
+  use_counter_service_workers_destroyed,
+  use_counter_shared_workers_destroyed,
+  use_counter_dedicated_workers_destroyed,
+  REGEXP_REPLACE(
+    REGEXP_REPLACE(
+      REGEXP_REPLACE(
+        REGEXP_REPLACE(
+          REGEXP_REPLACE(
+            REGEXP_REPLACE(
+              REGEXP_REPLACE(
+                REGEXP_REPLACE(
+                  REGEXP_REPLACE(metric, 'use_counter_css_doc_', 'use.counter.css.doc.'),
+                  'use_counter_css_page_',
+                  'use.counter.css.page.'
+                ),
+                'use_counter_worker_dedicated',
+                'use.counter.worker.dedicated.'
+              ),
+              'use_counter_worker_shared_',
+              'use.counter.worker.shared.'
```

Comment:
This seems like it could be a single regular expression? It's generally replacing `_` with `.` also looking for some specific strings.

Reply:
It can't be, because it needs to leave the dots in the latter half to match the glean dictionary names (otherwise I most definitely would've done that, that would have been sooo much easier)  :)